### PR TITLE
LPS-154947 Add new functionality to SystemProperties for moving module framework properties next

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian , @dantewang , This pr adds new functions, which are required when moving properties to system.Properties

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @zhangyan0701 , Could you please help me scan test results when you free ? Thanks!

**Comment by W2jx** [comment]
ci:test:acceptance

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:acceptance

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
Hi @zhangyan0701 , I updated the code and re-tested it。Could you please help me scan test results when you free ? Thanks!

**Comment by zhangyan0701** [comment]
No regressions found.

File: portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java [hunk 1]
```diff
@@ -100,6 +98,10 @@ public static Map<String, String> getProperties(
 		return properties;
 	}
 
+	public static Set<String> getPropertyNames() {
```
Comment (tinatian) [inline portal-kernel/src/com/liferay/portal/kernel/util/SystemProperties.java:101]: @W2jx The commit message is not consistent with the code change in this commit.

File: portal-kernel/src/com/liferay/portal/kernel/util/Validator.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
 package com.liferay.portal.kernel.util;
 
 import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringBundler;
```
Comment (tinatian) [inline portal-kernel/src/com/liferay/portal/kernel/util/Validator.java:18]: @W2jx This change is kind of weird, I know the root reason you did it, but let's try to focus on current task by not modifying kernel's StringBundler

**Comment by tinatian** [comment]
@W2jx Please resend.

Tina

File: portal-kernel/src/com/liferay/portal/kernel/util/Validator.java (hunk 2 above)
Comment (dantewang) [reply portal-kernel/src/com/liferay/portal/kernel/util/Validator.java:18]: @W2jx Additionally, please also skip other classes in portal-kernel that have copies in core/petra. They are going to be removed in the future.

**Pull request closed without merging**